src/Cardano/Wallet/Api/Types.hs

```diff
@@ -202,91 +199,111 @@ getApiMnemonicT (ApiMnemonicT (pw, _)) = pw
                              Encoding & Decoding
 -------------------------------------------------------------------------------}
 
-newtype DecodeApiAddressError = DecodeApiAddressError String
-    deriving stock Show
-    deriving newtype Buildable
+-- | Defines a textual encoding for a type defined within the API.
+class ToText a where
+    -- | Encode the specified value as text.
+    toText :: a -> Text
 
--- | Constructs an address from a Base58-encoded string.
---
--- Fails if the specified string is not Base58 encoded.
-decodeApiAddress :: Text -> Either DecodeApiAddressError (ApiT Address)
-decodeApiAddress x = maybe
-    (Left $ DecodeApiAddressError
-        "Unable to decode Address: expected Base58 encoding")
-    (pure . ApiT . Address)
-    (decodeBase58 bitcoinAlphabet $ T.encodeUtf8 x)
-
--- | Converts an address to a Base58-encoded string.
-encodeApiAddress :: ApiT Address -> Text
-encodeApiAddress =
-    T.decodeUtf8 . encodeBase58 bitcoinAlphabet . getAddress . getApiT
-
-newtype DecodeApiEncryptionPassphraseError
-    = DecodeApiEncryptionPassphraseError String
-    deriving stock Show
+-- | Defines a textual decoding for a type defined within the API.
+class FromText a where
+    -- | Decode the specified text as a value.
+    fromText :: Text -> Either TextDecodingError a
+
+-- | Indicates an error that occurred while decoding from text.
+newtype TextDecodingError = TextDecodingError
+    { getTextDecodingError :: String }
+    deriving stock (Eq, Show)
```

Comment:
Note the discussion here: https://github.com/input-output-hk/cardano-wallet/pull/133#discussion_r271592485